### PR TITLE
[AVS-585] feature-gate deps; rename rusoto module

### PR DESCRIPTION
**Pull request by scottlamb**
This is boring prep for using DynamoDB through the newer aws-sdk library without depending on both.

### CODE REVIEW

**Review by lzuosym — DISMISSED** [review]
Lgtm.

File: src/dynstore.rs [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
 #[async_trait]
 impl super::Backend for Backend {
     async fn get<'a, K: Into<Arg<'a>> + Send>(&self, key: K) -> Result<Option<Value>> {
-        match self {
-            Self::Memory(backend) => backend.get(key).await,
-            Self::Redis(backend) => backend.get(key).await,
-            Self::DynamoDB(backend) => backend.get(key).await,
-            Self::ReadCache(backend) => backend.get(key).await,
-        }
+        dispatch!(self, backend, { backend.get(key) })
```
Comment (ccbrown) [inline src/dynstore.rs:36]: Much nicer! :D

**Review by ccbrown — APPROVED** [review]
My only hesitation is I think it'd be a little clearer to keep "DynamoDB" in the variant / module names (e.g. "RusotoDynamoDB"), but I can go either way.

File: src/dynstore.rs (hunk 1 above)
Comment (scottlamb) [reply src/dynstore.rs:37]: > My only hesitation is I think it'd be a little clearer to keep "DynamoDB" in the variant / module names (e.g. "RusotoDynamoDB"), but I can go either way.

I debated this a bit myself and first chose brevity, but you're right: this doesn't get spelled that often, and it's better to be clear. Now says `rusoto_dynamodbstore` / `RusotoDynamoDB`. Coming soon: `aws_sdk_dynamodbstore` / `AwsSdkDynamoDB`. PTAL?

**Review by ccbrown — APPROVED** [review]

**Review by lzuosym — APPROVED** [review]

**Pull request merged**